examples/advanced_examples/plot_grid_search_withinsession.py

```diff
@@ -0,0 +1,97 @@
+"""
+============================
+GridSearch within a session
+============================
+
+This example shows how to use GridSearchCV within a session.
+
+"""
+import os
+
+from pyriemann.estimation import Covariances
+from pyriemann.spatialfilters import CSP
+from pyriemann.tangentspace import TangentSpace
+from sklearn.discriminant_analysis import LinearDiscriminantAnalysis as LDA
+from sklearn.linear_model import LogisticRegression
+from sklearn.pipeline import Pipeline
+
+from moabb.datasets import Zhou2016
+from moabb.evaluations import WithinSessionEvaluation
+from moabb.paradigms import MotorImagery
+
+
+sub_numb = 1
+
+# Initialize parameter for the Band Pass filter
+fmin = 8
+fmax = 35
+tmin = 0
+tmax = None
+
+# Select the Subject
+subjects = [int(sub_numb)]
+# Load the dataset, right now you have added Nothing events to DATA using new stim channel STI
+dataset = Zhou2016()
+
+events = ["right_hand", "feet"]
+
+paradigm = MotorImagery(
+    events=events, n_classes=len(events), fmin=fmin, fmax=fmax, tmax=tmax
+)
+
+# Create a path and folder for every subject
+path = os.path.join(str("try_Subject_" + str(sub_numb)))
+os.makedirs(path, exist_ok=True)
+
+# Pipelines
+pipelines = {}
+# Define the different algorithm to test and assign a name in the dictionary
+pipelines["CSP+LDA"] = Pipeline(
+    steps=[
+        ("Covariances", Covariances("cov")),
+        ("csp", CSP(nfilter=6)),
+        ("lda", LDA(solver="lsqr", shrinkage="auto")),
+    ]
+)
```

Comment:
```suggestion
pipelines["VanillaEN"] = Pipeline(
    steps=[
        ("Covariances", Covariances("cov")),
        ("Tangent_Space", TangentSpace(metric="riemann")),
        ("LogistReg", LogisticRegression(penalty="elasticnet",solver="saga")),
    ]
)
```
Maybe you could compare Cov+EN with grid search and "vanilla" Cov+EN, seems more appropriate to demonstrate the interest of grid search.